esx_service/utils/auth_data.py

```diff
@@ -374,6 +376,58 @@ def create_default_privileges(self):
             err = error_code.TENANT_SET_ACCESS_PRIVILEGES_FAILED.format(auth.DEFAULT_TENANT, auth.DEFAULT_DS, error_info)
             logging.warning(err)
 
+    def get_db_version(self):
+        """
+            Get DB schema version from auth-db
+        """
+        major_ver = 0
+        minor_ver = 0
+
+        # check table "versions" exist or not
+        # if table "versions" does not exist, return major_ver=0 and minor_ver=0
+        try:
+            cur = self.conn.execute("SELECT name FROM sqlite_master WHERE type = 'table' and name = 'versions';")
+            result = cur.fetchall()
+        except sqlite3.Error as e:
+            logging.error("Error %s when checking whether table versions exists or not", e)
+            return str(e), major_ver, minor_ver
+
+        if not result:
+        # table "versions" does not exist
+            return None, major_ver, minor_ver
+        
+        try:
+            cur = self.conn.execute("SELECT * FROM versions WHERE id = 0 ")
+            result = cur.fetchall()
+        except sqlite3.Error as e:
+            logging.error("Error %s when querying from table tenants versions", e)
+            return str(e), major_ver, minor_ver    
+        
+        major_ver = result[0][1]
+        minor_ver = result[0][2]
+        logging.debug("get_db_version: major_ver=%d minor_ver=%d", major_ver, minor_ver)
+        return None, major_ver, minor_ver
+
+    def need_upgrade_db(self):
+        """
+            Check auth-db schema need to be upgraded or not
+        """
+        major_ver = 0
+        minor_ver = 0
+        error_info, major_ver, minor_ver = self.get_db_version()
+        if error_info:
+            logging.error("need_upgrade_db: fail to get version info of auth-db, cannot do upgrade")
+            return False
+        
+        if major_ver != DB_MAJOR_VER or minor_ver != DB_MINOR_VER:
+            auth_db_ver = get_version_str(major_ver, minor_ver)
+            curr_db_ver = get_version_str(DB_MAJOR_VER, DB_MINOR_VER)
+            logging.error("version %s in auth-db does not match latest DB version %s",
```

Comment:
need some info on what to do next :-). 
E.g. "DB upgrade is not supported. Please remove the DB and create new configuration using <admin cmd line>"